Python/Product/PythonTools/PythonTools/Extensions.cs

```diff
@@ -272,39 +272,15 @@ out guid
             return PythonLanguageVersion.None;
         }
 
-        internal static IPythonInterpreterFactory GetInterpreterFactoryAtCaret(this ITextView textView, IServiceProvider serviceProvider) {
+        internal static InterpreterConfiguration GetInterpreterConfigurationAtCaret(this ITextView textView, IServiceProvider serviceProvider) {
             var client = PythonLanguageClient.FindLanguageClient(textView.TextBuffer);
-            if (client?.Factory != null) {
-                return client.Factory;
+            if (client?.Configuration != null) {
+                return client.Configuration;
             }
 
-            return serviceProvider.GetPythonToolsService().InterpreterOptionsService.DefaultInterpreter;
-        }
-
-        internal static InterpreterConfiguration GetInterpreterConfigurationAtCaret(this ITextView textView, IServiceProvider serviceProvider) {
-            return textView.GetInterpreterFactoryAtCaret(serviceProvider)?.Configuration;
+            return serviceProvider.GetPythonToolsService().InterpreterOptionsService.DefaultInterpreter?.Configuration;
```

Comment:
Looks like GetInterpreterConfigurationAtCaret is called from an eventhandler..

should we add a switch to main thread here?

Reply:
Both callers of this method are from command handlers, so they are on the UI thread.

The call to `FindLanguageClient` is thread safe.

`GetPythonToolsService` throws in debug builds if called from anything but the UI thread.

Threading rules say that if this was an async method, we should do the switch to main thread. In a sync method, recommendation is to throw if you need to be on main thread but aren't on it, which `GetPythonToolsService` does, but it may take a while to discover that issue if most of the calls return early in the if statement.

I guess I'll add a throw at the top.